--- a/CLHEP/cmake/Modules/ClhepBuildLibrary.cmake
+++ b/CLHEP/cmake/Modules/ClhepBuildLibrary.cmake
@@ -38,10 +38,15 @@ macro(clhep_build_library package)
       )
   SET_TARGET_PROPERTIES(${package}S
       PROPERTIES 
-        OUTPUT_NAME CLHEP-${package}-${VERSION}
+        OUTPUT_NAME CLHEP-${package}S-${VERSION}
       )
 
-  target_link_libraries(${package}  ${package_library_list} )
+  if(WIN32)
+    # Avoid linkage error of broken shared libs on Windows
+    target_link_libraries(${package}  ${package_library_list_static} )
+  else()
+    target_link_libraries(${package}  ${package_library_list} )
+  endif()
   target_link_libraries(${package}S ${package_library_list_static} )
 
   # Install the libraries
@@ -71,7 +76,7 @@ macro(clhep_build_libclhep )
       )
   SET_TARGET_PROPERTIES(CLHEPS
       PROPERTIES 
-        OUTPUT_NAME CLHEP-${VERSION}
+        OUTPUT_NAME CLHEPS-${VERSION}
       )
 
   # export creates library dependency files for CLHEPConfig.cmake
